dartagnan/src/main/java/com/dat3m/dartagnan/wmm/Wmm.java

```diff
@@ -203,8 +203,17 @@ private Relation makePredefinedRelation(String name) {
             case CTRLDIRECT -> new DirectControlDependency(r);
             case EMPTY -> new Empty(r);
             case FR ->  {
-                Relation rfinv = addDefinition(new Inverse(newRelation(), getOrCreatePredefinedRelation(RF)));
-                yield composition(r, rfinv, getOrCreatePredefinedRelation(CO));
+                final Relation reads = addDefinition(new SetIdentity(newRelation(), getFilter(Tag.READ)));
+                final Relation rfRange = addDefinition(new RangeIdentity(newRelation(), getOrCreatePredefinedRelation(RF)));
+                final Relation ur = addDefinition(new Difference(newRelation(), reads, rfRange));
+                final Relation rfinv = addDefinition(new Inverse(newRelation(), getOrCreatePredefinedRelation(RF)));
+                //final Relation ur = getOrCreatePredefinedRelation(UR);
```

Comment:
This can go.
Also, it would be useful to add a documentation comment with the new definition of `fr` in plain cat (reading this programmatically is not straight forward)